src/freenet/node/OpennetManager.java

```diff
@@ -546,6 +589,39 @@ public boolean wantPeer(OpennetPeerNode nodeToAddNow, boolean addAtLRU, boolean
 				return false;
 			}
 		}
+		// Don't connect to peers in excluded countries
+		if (nodeToAddNow != null) {
+			Peer[] handshakeIPs = nodeToAddNow.getHandshakeIPs();
+			if (handshakeIPs != null) {
+				for (Peer peer : handshakeIPs) {
+					if (peer == null) {
+						continue;
+					}
+					FreenetInetAddress address = peer.getFreenetAddress();
+					if (address == null) {
+						continue;
+					}
+					InetAddress inetAddress = address.getAddress(false);
+
+					IPConverter ipConverter = IPConverter.getInstance(NodeFile.IPv4ToCountry.getFile(node));
+					try {
+						Country country = ipConverter.locateIP(inetAddress.getAddress());
+						if (country != null) {
+							if (excludedCountries.contains(country.toString().toUpperCase())) {
```

Comment:
Additionally it is inefficient to always to the toUpperCase() before each contains() query of the HashSet.
Instead the initialization of the HashSet should make the set contain the same case as which country.toString() returns, then case conversion only has to happen once at init, not at every query.